textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcher.java

```diff
@@ -34,22 +51,45 @@ public void open() throws DataFlowException {
 
     @Override
     public ITuple getNextTuple() throws DataFlowException {
-
-        try {
+		try {
             ITuple sourceTuple = sourceOperator.getNextTuple();
             if(sourceTuple == null){
                 return null;
             }
-            if (predicate.satisfy(sourceTuple)) {
-                return sourceTuple;
-            } else {
-                return getNextTuple();
+            
+            RegexPredicate regexPredicate = (RegexPredicate)predicate; 
+            
+            spans = regexPredicate.computeMatches(sourceTuple);
+            
+            if (spans != null && spans.size() != 0) { // a list of matches found
+            	if (sourceTupleSchema == null) {
+            		sourceTupleSchema = sourceTuple.getSchema();
+            	}
+            	if (spanSchema == null) {
+            		spanSchema = Utils.createSpanSchema(sourceTupleSchema);
+            	}
+            	fields = sourceTuple.getFields();
+            	return constructSpanTuple(spans);
+            } else { // no match found
+            	return getNextTuple();
             }
         } catch (Exception e) {
             e.printStackTrace();
             throw new DataFlowException(e.getMessage(), e);
-        }
-
+        }        
+    }
+    
+    private ITuple constructSpanTuple(List<Span> spans) {
```

Comment:
@laisycs Sorry for the delay. Please use the method `Utils.getSpanTuple()` instead of a new method `constructSpanTuple`. You can do this in the next PR.